homeassistant/util/logging.py

```diff
@@ -119,28 +119,28 @@ def log_exception(format_err: Callable[..., Any], *args: Any) -> None:
 @overload
 def catch_log_exception(
     func: Callable[..., Coroutine[Any, Any, Any]], format_err: Callable[..., Any]
-) -> Callable[..., Coroutine[Any, Any, None]]:
+) -> Callable[..., Coroutine[Any, Any, None]] | Callable[..., None]:
     """Overload for Callables that return a Coroutine."""
 
 
 @overload
 def catch_log_exception(
     func: Callable[..., Any], format_err: Callable[..., Any]
-) -> Callable[..., None | Coroutine[Any, Any, None]]:
+) -> Callable[..., Coroutine[Any, Any, None]] | Callable[..., None]:
     """Overload for Callables that return Any."""
 
 
 def catch_log_exception(
     func: Callable[..., Any], format_err: Callable[..., Any]
-) -> Callable[..., None | Coroutine[Any, Any, None]]:
+) -> Callable[..., Coroutine[Any, Any, None]] | Callable[..., None]:
     """Decorate a callback to catch and log exceptions."""
```

Comment:
```suggestion
    """Decorate a Callable to catch and log exceptions."""
```

Reply:
> 

Is this a better docstr?

Reply:
Yeah, but maybe be more precis about the return types:
"""Decorate a function func to catch and log exceptions.

    If func is a coroutine function, a coroutine function will be returned.
    If func is a callback, a callback will be returned.
"""